numpy/core/src/multiarray/einsum.c.src

```diff
@@ -45,6 +56,21 @@
 #include <emmintrin.h>
 #endif
 
+#if EINSUM_USE_NEON_32 || EINSUM_USE_NEON_64
+#include <arm_neon.h>
+static void prefetch(const double *ptr, size_t offset)
```

Comment:
you may need to add a new intrin for prefetch or you can keep it as-is,
discover the SIMD extensions [dir](https://github.com/numpy/numpy/tree/master/numpy/core/src/common/simd) to see how we do it.